src/Unleash/Communication/UnleashApiClient.cs

```diff
@@ -233,6 +234,7 @@ public async Task<bool> SendMetrics(ThreadSafeMetricsBucket metrics, Cancellatio
                     InstanceId = clientRequestHeaders.InstanceTag,
                     Bucket = bucket
                 });
+                memoryStream.Position = 0;
```

Comment:
Should we also update the docs?

Also, what we discussed is why should we allow people to override the json serialization mechanism? Maybe that feature should be deprecated

Reply:
> Should we also update the docs?

Yes, I'll amend this PR

> Also, what we discussed is why should we allow people to override the json serialization mechanism? Maybe that feature should be deprecated

Absolutely, but feels like a bigger question/release since we're changing the public API. I figured this would be a small fix in the meantime while we wrap up what we're working on for the next major release (Ygg++)

Do you agree?

Reply:
Agree!